pydv/pydvpy.py

```diff
@@ -437,146 +437,98 @@ def read(fname, gnu=False, xcol=0, verbose=False, pattern=None, matches=None):
+
+    curve_list = list()
     regex = None
-    matchcount = 0
 
-    if pattern is not None:
+    if pattern:
         regex = re.compile(r"%s" % pattern)
-
+    file_name = os.path.expanduser(file_name)
+    _, ext = os.path.splitext(file_name)
     try:
-        if fname.strip()[0] == '~':
-            fname = os.getenv('HOME') + fname.strip()[1:]
-
-        if gnu or fname[-4:] == '.gnu':
-            return __loadcolumns(fname, xcol)
+        if gnu or ext == '.gnu':
+            return __loadcolumns(file_name, xcol)
 
         if pdbLoaded:
             try:
-                fpdb = pdb.open(fname, 'r')
-                return __loadpdb(fname, fpdb)
+                fpdb = pdb.open(file_name, 'r')
+                return __loadpdb(file_name, fpdb)
             except:
                 pass
 
-        f = open(fname, 'r')
-        current = None # curve.Curve(fname, "Unlabeled")
-        first = 1
-        buildlistx = list()
-        buildlisty = list()
-
-        for line in f:
-            if line.strip()[:2] == '##':
-                continue
-
-            if line.strip()[:1] == '#':  # check for start of new curve
-                if matches is not None:
-                    # print "matches = %d" % matches
-                    if matchcount >= matches:
-                        break
-
-                if first != 1:
-                    if len(buildlistx) != 0:
-                        if current.drawstyle != 'default':
-                            current.drawstyle = 'default'
-                            buildlisty.append(buildlisty[-1])
-                            histx = list()
-                            histx.append(buildlistx[0])
-                            histy = list()
-                            histy.append(buildlisty[0])
-
-                            i = 1
-                            while i < len(buildlisty):
-                                # insert new x,y
-                                histx.append(buildlistx[i])
-                                histy.append(buildlisty[i - 1])
-
-                                # insert old x,y
-                                histx.append(buildlistx[i])
-                                histy.append(buildlisty[i])
-
-                                i += 1
-
-                            current.x = np.array(histx)
-                            current.y = np.array(histy)
-                        else:
-                            current.x = np.array(buildlistx)
-                            current.y = np.array(buildlisty)
-
-                        curvelist.append(current)
-                        buildlistx = list()
-                        buildlisty = list()
+        match_count = 0
+        build_list_x = list()
+        build_list_y = list()
+        current = None
+        new_curve = True
+        potential_curve_name = ""
+        with open(file_name, 'r') as f:
+            for line in f:
+                split_line = re.split(r'[ _]+', str.strip(line))
```

Comment:
@fillmore1 I had to remove the map so that I could use `re` to split the line since I needed a bit more logic than what `str.split` provides. Not sure if this affects the performance improvement you were hoping for, but give it at try and we'll see if we need to make anymore changes.